packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -155,8 +157,7 @@ function transform(codeGen: CodeGen): t.Expression {
             } else if (isComment(child) && codeGen.preserveComments) {
                 res.push(transformComment(child));
             } else if (isIfBlock(child)) {
-                const children = transformIfBlock(child);
-                Array.isArray(children) ? res.push(...children) : res.push(children);
+                res.push(transformIfBlock(child));
             } else {
                 // Todo: create validation for case when there's invalid AST nodes
```

Comment:
Let's add a comment here that ElseifBlock and ElseBlock are two AST nodes that we should never see in the `children` array property. They should always be attached to the `else` property.